### PR TITLE
Fix menu not closed on blur

### LINKED ISSUE #575: CSS value input dropdown is not closed after changing focus
1 place cursor into input
2 press down to open combobox dropdown
3 press tab to edit next field
4 press down to open another combobox dropdown

image

### PR DESCRIPTION
**Pull request by istarkov**
closes #575

## Before requesting a review

- [ ] if the PR is WIP - use draft mode
- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")
- [ ] what kind of review is needed?
  - [ ] conceptual
  - [ ] detailed
  - [ ] with testing

## Test cases

- [ ] step by step interaction description and what is expected to happen

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-designer/blob/main/apps/designer/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `designer/env-check.js` if mandatory

### CODE REVIEW

File: apps/designer/app/designer/features/style-panel/shared/css-value-input/css-value-input.tsx [hunk 1]
```diff
@@ -309,13 +309,13 @@ export const CssValueInput = ({
   });
 
   const handleOnBlur: KeyboardEventHandler = (event) => {
+    inputProps.onBlur(event);
     // When select is open, onBlur is triggered,though we don't want a change event in this case.
-    if (isUnitsOpen || isOpen) {
+    if (isUnitsOpen || (isOpen && !getMenuProps().empty)) {
```
Comment (istarkov) [inline apps/designer/app/designer/features/style-panel/shared/css-value-input/css-value-input.tsx:314]: MB anyone knows isOpen is always true when we are editing input. Independently does menu has items or not,
is it solvable in downshift?

Comment (istarkov) [reply apps/designer/app/designer/features/style-panel/shared/css-value-input/css-value-input.tsx:314]: Looks like strange default behavior,  semantically its isFocused

File: apps/designer/app/designer/features/style-panel/shared/css-value-input/css-value-input.tsx [hunk 2]
```diff
@@ -309,13 +309,13 @@ export const CssValueInput = ({
...
       return;
     }
 
     onChangeComplete(value);
-    inputProps.onBlur(event);
```
Comment (kof) [inline apps/designer/app/designer/features/style-panel/shared/css-value-input/css-value-input.tsx:318]: that's strange, I feel like this was needed

Comment (kof) [reply apps/designer/app/designer/features/style-panel/shared/css-value-input/css-value-input.tsx:318]: ah you moved it

**Review by kof — APPROVED** [review]

File: apps/designer/app/designer/features/style-panel/shared/css-value-input/css-value-input.tsx (hunk 1 above)
Comment (kof) [reply apps/designer/app/designer/features/style-panel/shared/css-value-input/css-value-input.tsx:314]: and yes the dropdown may be not empty even when not shown

Comment (rpominov) [reply apps/designer/app/designer/features/style-panel/shared/css-value-input/css-value-input.tsx:314]: > When select is open, onBlur is triggered

Maybe that's where we broke something? In codesanbox above input does not loose focus when downshift opens

Comment (istarkov) [reply apps/designer/app/designer/features/style-panel/shared/css-value-input/css-value-input.tsx:314]: Here 2 comboboxes. Comment is for 2nd. 
Will split on 2 ifs with different comments

Comment (istarkov) [reply apps/designer/app/designer/features/style-panel/shared/css-value-input/css-value-input.tsx]: Splitted

**Pull request merged**